### PR TITLE
Add support for one-peer peersets

### PR DESCRIPTION
**Pull request by kjarosh**

### CODE REVIEW

File: src/main/kotlin/com/github/davenury/ucac/consensus/raft/infrastructure/RaftConsensusProtocolImpl.kt [hunk 1]
```diff
@@ -504,11 +504,13 @@ class RaftConsensusProtocolImpl(
...
     private suspend fun applyAcceptedChanges() {
+        applyAcceptedChanges(voteContainer.getAcceptedChanges { isMoreThanHalf(it) })
+    }
 
+    private suspend fun applyAcceptedChanges(acceptedIds: List<String>) {
         val acceptedItems: List<LedgerItem>
 
         mutex.withLock {
-            val acceptedIds: List<String> = voteContainer.getAcceptedChanges { isMoreThanHalf(it) }
             acceptedItems = state.getLogEntries(acceptedIds)
```
Comment (Rados13) [inline src/main/kotlin/com/github/davenury/ucac/consensus/raft/infrastructure/RaftConsensusProtocolImpl.kt:514]: This modification isn't thread-safe.

**Comment by Davenury** [comment]
Soooo.... we're going to completely ignore GPAC which wasn't tested on such case?

File: src/main/kotlin/com/github/davenury/ucac/consensus/raft/infrastructure/RaftConsensusProtocolImpl.kt (hunk 1 above)
Comment (kjarosh) [reply src/main/kotlin/com/github/davenury/ucac/consensus/raft/infrastructure/RaftConsensusProtocolImpl.kt]: Fixed

**Comment by kjarosh** [comment]
Added support for one-peer peersets for AC too

**Review by Rados13 — APPROVED** [review]

**Pull request merged**